src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -695,6 +701,84 @@ class spell_mage_water_elemental_freeze : public SpellScript
     }
 };
 
+// Ice Lance - 30455
+class spell_mage_ice_lance : public SpellScript
+{
+    PrepareSpellScript(spell_mage_ice_lance);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({
+            SPELL_MAGE_ICE_LANCE,
+            SPELL_MAGE_ICE_LANCE_TRIGGER
+        });
+    }
+
+    void HandleOnHit(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        Unit* target = GetHitUnit();
+        if (!caster && !target)
+            return;
+
+        bool targetFrozen = target->HasAuraState(AURA_STATE_FROZEN);
+
+        // Thermal Void
+        if (caster->HasAura(SPELL_MAGE_THERMAL_VOID) && (targetFrozen || caster->HasAura(SPELL_MAGE_FINGERS_OF_FROST)))
+        {
+            if (Aura* icyVeins = caster->GetAura(SPELL_MAGE_ICY_VEINS))
+            {
+                if (SpellInfo const* thermalVoidInfo = sSpellMgr->GetSpellInfo(SPELL_MAGE_THERMAL_VOID, GetCastDifficulty()))
+                {
+                    if (SpellEffectInfo const* eff0 = thermalVoidInfo->GetEffect(EFFECT_0))
+                    {
+                        int32 increaseDuration = eff0->CalcValue() * IN_MILLISECONDS;
+                        int32 newDuration = (icyVeins->GetDuration() + increaseDuration);
+                        icyVeins->SetDuration(newDuration);
+                    }
+                }
+            }
+        }
+
+        // if( target->HasAuraState(AURA_STATE_FROZEN))
+            caster->CastSpell(target, SPELL_MAGE_ICE_LANCE_TRIGGER, true); // TODO damage x3 on frozen target
+
+        // Splitting Ice
+        if (caster->HasAura(SPELL_MAGE_SPLITTING_ICE))
```

Comment:
You don't need to handle Splitting Ice at all. That talent applies a spell modifier (JUMP_TARGETS) that makes 228598 cleave automatically using chain target selection (like chain lightning) with a modified damage using ChainAmplitude column from SpellEffect.db2